frontend/libs/studio-components/src/components/StudioExpression/SimplifiedEditor/LogicalExpressionEditor/SubExpression/Subexpression.tsx

```diff
@@ -26,10 +27,13 @@ export type SubexpressionProps = {
 };
 
 export const Subexpression = ({ expression, legend, onChange, onDelete }: SubexpressionProps) => {
-  const { texts } = useStudioExpressionContext();
+  const { texts, dataLookupOptions } = useStudioExpressionContext();
   const [isInEditMode, setIsInEditMode] = useState<boolean>(false);
   const [expressionState, setExpressionState] = useState<SimpleSubexpression>(expression);
-  const [errors, setErrors] = useState<ExpressionErrorKey[]>([]);
+  const componentOptions = dataLookupOptions[DataLookupFuncName.Component];
```

Comment:
If I understand the code correctly - the componentOptions lookup will always happen regardless of the expression is a component lookup or not? 🤔